tensorzero-core/src/providers/openai/responses.rs

```diff
@@ -0,0 +1,717 @@
+            ContentBlock::Text(Text { text }) => {
+                messages.push(OpenAIResponsesInput::Message(OpenAIResponsesInputMessage {
+                    id: None,
+                    role: "user",
+                    content: vec![OpenAIResponsesInputMessageContent::InputText {
+                        text: Cow::Borrowed(text),
+                    }],
+                }));
+            }
+            ContentBlock::ToolCall(_) => {
+                return Err(Error::new(ErrorDetails::InvalidMessage {
+                    message: "Tool calls are not supported in user messages".to_string(),
+                }));
+            }
+            ContentBlock::ToolResult(tool_result) => {
+                messages.push(OpenAIResponsesInput::FunctionCallOutput(
+                    OpenAIResponsesFunctionCallOutput {
+                        output: Cow::Borrowed(&tool_result.result),
+                        call_id: Cow::Borrowed(&tool_result.id),
+                    },
+                ));
+            }
+            ContentBlock::File(file) => {
+                let content_block = prepare_file_message(file, messages_config).await?;
+                match content_block {
+                    OpenAIContentBlock::ImageUrl { image_url } => {
+                        messages.push(OpenAIResponsesInput::Message(OpenAIResponsesInputMessage {
+                            id: None,
+                            role: "user",
+                            content: vec![OpenAIResponsesInputMessageContent::InputImage {
+                                image_url: Cow::Owned(image_url.url),
+                            }],
+                        }));
+                    }
+                    OpenAIContentBlock::File { file } => {
+                        messages.push(OpenAIResponsesInput::Message(OpenAIResponsesInputMessage {
+                            id: None,
+                            role: "user",
+                            content: vec![OpenAIResponsesInputMessageContent::InputFile { file }],
+                        }));
+                    }
+                    _ => {
+                        return Err(Error::new(ErrorDetails::InternalError {
+                            message: format!("`prepare_file_message` produced an unexpected content block. {IMPOSSIBLE_ERROR_MESSAGE}")
+                        }));
+                    }
+                }
+            }
+            ContentBlock::Thought(thought) => {
+                // TODO - figure out how to handle this correctly
+                warn_discarded_thought_block(messages_config.provider_type, thought);
+            }
+            ContentBlock::Unknown {
+                data,
+                model_provider_name: _,
+            } => {
+                messages.push(OpenAIResponsesInput::Message(OpenAIResponsesInputMessage {
+                    id: None,
+                    role: "user",
+                    content: vec![OpenAIResponsesInputMessageContent::Unknown {
```

Comment:
I don't think I understand this; can you add a comment explaining why this behavior is correct?

Reply:
Updated